qcodes/tests/test_parameter.py

```diff
@@ -223,6 +235,124 @@ def test_get_latest(self):
         self.assertEqual(local_parameter.get_latest(), 2)
         self.assertGreater(local_parameter.get_latest.get_timestamp(), after_set)
 
+    def test_get_latest_unknown(self):
+        """
+        Test that get latest on a parameter that has not been acquired will
+        trigger a get
+        """
+        value = 1
+        local_parameter = BetterGettableParam('test_param', set_cmd=None,
+                                              get_cmd=None)
+        # fake a parameter that has a value but never been get/set to mock
+        # an instrument.
+        local_parameter._latest = {"value": value, "raw_value": value,
+                                   'ts': None}
+        assert local_parameter.get_latest.get_timestamp() is None
+        before_get = datetime.now()
+        assert local_parameter._get_count == 0
+        assert local_parameter.get_latest() == value
+        assert local_parameter._get_count == 1
+        # calling get_latest above will call get since TS is None
+        # and the TS will therefore no longer be None
+        assert local_parameter.get_latest.get_timestamp() is not None
+        assert local_parameter.get_latest.get_timestamp() >= before_get
```

Comment:
Shall we make this stricter by using the sleep for time resolution?

Reply:
I would prefer not to. We have been bitten by time resolution introduced failures before hand. This is enough to test what we want